shardy/dialect/sdy/ir/attrs.td

```diff
@@ -738,6 +779,19 @@ def Sdy_OpShardingRule : AttrDef<Sdy_Dialect, "OpShardingRule"> {
     these ops, so a user must tell it how. When it is a custom rule, then the
     rule is always preserved/never removed. `is_custom_rule` can only be true
     for `stablehlo.custom_call` ops.
+
+    Constraints:
+
+    - Number of mappings and types match.
```

Comment:
We can't say types without giving context as this isn't a field of the attribute, we can say - number of operand/result mappings must match the number of operands/results of the op,